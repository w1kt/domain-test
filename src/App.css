@import-normalize;
@import url('https://fonts.googleapis.com/css?family=Merriweather&display=swap');
@import url('https://fonts.googleapis.com/css?family=Merriweather+Sans:400,700,800&display=swap');

.App {
  font-family: 'Merriweather Sans', serif;
  display: flex;
  flex: 1;
  max-width: 1000px;
}
.App-mockup {
  opacity: 0.3;
  position: absolute;
}
h1,
h2 {
  margin: 0;
}
* {
  box-sizing: border-box
}
