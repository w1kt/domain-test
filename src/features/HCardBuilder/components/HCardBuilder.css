.HCardBuilder {
  padding: 40px;
  display: flex;
  flex: 0.767;
  flex-direction: column;
  box-sizing: border-box;
}
.flex-grid {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  margin: 0 -10px;
}
.flex-grid.vertical-margin {
  margin: -7.5px -10px;
}
.flex-grid.two-col > * {
  display: flex;
  flex-direction: column;
  width: calc(50% - 20px);
}
.flex-grid > * {
  margin: 0 10px;
}
.flex-grid.vertical-margin > * {
  margin: 7.5px 10px;
}