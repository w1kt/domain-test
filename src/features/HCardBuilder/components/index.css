.HCardBuilder {
  padding: 40px;
  width: 400px
}
.flex-grid {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  margin: 0 -10px;
}
.flex-grid.vertical-margin {
  margin: -7.5px -10px;
}
.flex-grid.two-col > * {
  display: flex;
  flex-direction: column;
  flex-basis: calc(50% - 20px);
}
.flex-grid > * {
  margin: 0 10px;
}
.flex-grid.vertical-margin > * {
  margin: 7.5px 10px;
}