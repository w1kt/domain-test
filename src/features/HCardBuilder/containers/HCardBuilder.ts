import { connect } from 'react-redux';
import { ConfigProps, DecoratedComponentClass } from 'redux-form';
import { State } from '../../../reducers/rootReducer';
import { setAvatar } from '../actions/actions';
import HCardBuilderForm from '../components/HCardBuilder';

type OwnProps = DecoratedComponentClass<
  {},
  Partial<ConfigProps<{}, {}, string>>,
  string
>;

const mapStateToProps = (state: State, ownProps: OwnProps) => ({});
const mapStateToDispatch = {
  setAvatar
};

export default connect(
  mapStateToProps,
  mapStateToDispatch
)(HCardBuilderForm);

export type HCardBuilderProps = ReturnType<typeof mapStateToProps> &
  typeof mapStateToDispatch
