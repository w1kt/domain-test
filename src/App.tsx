import React, { PureComponent } from 'react';
import './App.css';
import HCardBuilder from './features/HCardBuilder/containers/HCardBuilder';

export class App extends PureComponent {
  public render() {
    return (
      <div className="App">
        <img className="App-mockup" src="./completed-view.png"></img>
        <HCardBuilder
          onSubmit={vals => console.log('Sending to backend', vals)}
        />
      </div>
    );
  }
}

export default App;
