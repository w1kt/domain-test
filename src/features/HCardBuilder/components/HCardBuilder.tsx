import React, { Component, PureComponent } from 'react';
import { reduxForm, InjectedFormProps } from 'redux-form';
import { HCardBuilderProps } from '../containers/HCardBuilder';
import CreateHCardButton from './CreateHCardButton/CreateHCardButton';
import FormGroup from './FormGroup/FormGroup';
import './HCardBuilder.css';
import TextFormField from './TextFormField/TextFormField';
import Title from './Title/Title';
import UploadAvatarButton from './UploadAvatarButton/UploadAvatarButton';
import { UserDetails } from './UserDetails';

/**
 * This component shows a form that allows users to input their details in order
 * to build a hCard.
 */
class HCardBuilder extends PureComponent<
  InjectedFormProps<UserDetails, HCardBuilderProps> & HCardBuilderProps
> {
  public render() {
    return (
      <form className="HCardBuilder" onSubmit={this.props.handleSubmit}>
        <Title text="hCard Builder"></Title>
        <FormGroup groupTitle="Personal Details">
          <TextFormField label="Given Name" />
          <TextFormField label="Surname" />
          <TextFormField label="Email" />
          <TextFormField label="Phone" />
        </FormGroup>
        <FormGroup groupTitle="Address">
          <TextFormField label="House Name or #" />
          <TextFormField label="Street" />
          <TextFormField label="Suburb" />
          <TextFormField label="State" />
          <TextFormField label="Postcode" />
          <TextFormField label="Country" />
        </FormGroup>
        <div className="flex-grid two-col">
          <UploadAvatarButton setAvatar={this.props.setAvatar} />
          <CreateHCardButton />
        </div>
      </form>
    );
  }
}

const HCardBuilderForm = reduxForm<UserDetails, HCardBuilderProps>({
  form: 'hCardBuilder'
})(HCardBuilder);

export default HCardBuilderForm;
