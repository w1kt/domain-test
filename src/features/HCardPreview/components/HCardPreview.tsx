import React from 'react';
import ComponentPreview from '../../ComponentPreview/components/ComponentPreview';
import HCard from '../../HCard/components/HCard';
import { HCardPreviewProps } from '../containers/HCardPreview';

/**
 *  Created a preview component and gives it a hCard as a subject.
 *  The hCard is fed its props from the form state in the store.
 */
const HCardPreview: React.FC<HCardPreviewProps> = props => {
  return (
    <ComponentPreview name="hCard Preview">
      <HCard {...props} />
    </ComponentPreview>
  );
};

export default React.memo(HCardPreview);
