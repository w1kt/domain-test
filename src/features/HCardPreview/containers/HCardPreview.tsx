import { State } from '../../../reducers/rootReducer';
import { connect } from 'react-redux';
import HCardPreview from '../components/HCardPreview';
import { UserDetails } from '../../HCardBuilder/components/UserDetails';

const mapStateToProps = (state: State) => {
  const formState = state.form.hCardBuilder;
  let res = {
    avatarUrl: state.hCardBuilder.avatarUrl
  }
  if (!formState) {
    return res;
  }
  const userDetails = formState.values as Partial<UserDetails>;
  return {
    ...res,
    ...userDetails
  }
};

const mapStateToDispatch = {};

export default connect(
  mapStateToProps,
  mapStateToDispatch
)(HCardPreview);

export type HCardPreviewProps = ReturnType<typeof mapStateToProps> &
  typeof mapStateToDispatch;
