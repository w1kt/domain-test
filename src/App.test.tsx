import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import HCardBuilder from './features/HCardBuilder/containers/HCardBuilder';
import HCardPreview from './features/HCardPreview/containers/HCardPreview';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const setup = () => {
  const props = {};
  const wrapper = shallow(<App />);
  return { props, wrapper };
};

it('renders without crashing', () => {
  const { wrapper } = setup();
  expect(wrapper).toBeDefined();
});

it('renders self and subcomponents', () => {
  const { wrapper } = setup();
  expect(
    wrapper
      .find('div')
      .first()
      .hasClass('App')
  ).toBe(true);
  expect(wrapper.containsMatchingElement(<HCardBuilder />));
  expect(wrapper.containsMatchingElement(<HCardPreview />));
});

it('renders correctly', () => {
  const mockStore = configureMockStore();
  const store = mockStore({ hCardBuilder: { avatarUrl: '' } });

  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
