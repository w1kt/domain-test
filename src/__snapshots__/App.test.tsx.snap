// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <form
    className="HCardBuilder"
    onSubmit={[Function]}
  >
    <h1
      className="Title"
      style={
        Object {
          "color": "#394F63",
        }
      }
    >
      hCard Builder
    </h1>
    <div
      className="FormGroup"
    >
      <h2
        className="GroupTitle"
      >
        Personal Details
      </h2>
      <div
        className="flex-grid two-col vertical-margin"
      >
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="given-name"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            Given Name
          </label>
          <input
            className="TextInput"
            id="given-name"
            name="given-name"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="surname"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            Surname
          </label>
          <input
            className="TextInput"
            id="surname"
            name="surname"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="email"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            Email
          </label>
          <input
            className="TextInput"
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="phone"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            Phone
          </label>
          <input
            className="TextInput"
            id="phone"
            name="phone"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="FormGroup"
    >
      <h2
        className="GroupTitle"
      >
        Address
      </h2>
      <div
        className="flex-grid two-col vertical-margin"
      >
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="house-name-or-#"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            House Name or #
          </label>
          <input
            className="TextInput"
            id="house-name-or-#"
            name="house-name-or-#"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="street"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            Street
          </label>
          <input
            className="TextInput"
            id="street"
            name="street"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="suburb"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            Suburb
          </label>
          <input
            className="TextInput"
            id="suburb"
            name="suburb"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="state"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            State
          </label>
          <input
            className="TextInput"
            id="state"
            name="state"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="postcode"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            Postcode
          </label>
          <input
            className="TextInput"
            id="postcode"
            name="postcode"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FieldLabel"
            htmlFor="country"
            style={
              Object {
                "color": "#394F63",
              }
            }
          >
            Country
          </label>
          <input
            className="TextInput"
            id="country"
            name="country"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="flex-grid two-col"
    >
      <div
        className="Button"
        style={
          Object {
            "backgroundColor": "#758E9D",
            "borderColor": "#506470",
          }
        }
      >
        <label
          htmlFor="files-input"
        >
          Upload Avatar
        </label>
        <input
          accept="image/*"
          id="files-input"
          onChange={[Function]}
          type="file"
        />
      </div>
      <button
        className="Button"
        style={
          Object {
            "backgroundColor": "#3FA9E2",
            "borderColor": "#3193C6",
          }
        }
        type="submit"
      >
        Create hCard
      </button>
    </div>
  </form>
  <div
    className="ComponentPreview"
  >
    <div
      className="preview-content"
    >
      <h2
        className="preview-name"
      >
        hCard Preview
      </h2>
      <div
        className="HCard"
      >
        <div
          className="HeaderBar"
          style={
            Object {
              "backgroundColor": "#394F63",
            }
          }
        >
          <h1
            className="title"
          >
             
          </h1>
          <img
            alt="User Avatar"
            className="Avatar"
            src="./default-avatar.jpg"
          />
        </div>
        <div
          className="hcard-inner"
        >
          <div
            className="CardItem "
          >
            <div
              className="card-item-inner"
            >
              <label
                className="card-item-label"
                htmlFor="card-item-value"
                style={
                  Object {
                    "color": "#394F63",
                  }
                }
              >
                Email
              </label>
              <span
                className="card-item-value"
                id="card-item-value"
              >
                
              </span>
            </div>
          </div>
          <div
            className="CardItem "
          >
            <div
              className="card-item-inner"
            >
              <label
                className="card-item-label"
                htmlFor="card-item-value"
                style={
                  Object {
                    "color": "#394F63",
                  }
                }
              >
                Phone
              </label>
              <span
                className="card-item-value"
                id="card-item-value"
              >
                
              </span>
            </div>
          </div>
          <div
            className="CardItem "
          >
            <div
              className="card-item-inner"
            >
              <label
                className="card-item-label"
                htmlFor="card-item-value"
                style={
                  Object {
                    "color": "#394F63",
                  }
                }
              >
                Address
              </label>
              <span
                className="card-item-value"
                id="card-item-value"
              >
                 
              </span>
            </div>
            <span
              className="card-item-extra-value"
            >
              
            </span>
          </div>
          <div
            className="CardItem inline"
          >
            <div
              className="card-item-inner"
            >
              <label
                className="card-item-label"
                htmlFor="card-item-value"
                style={
                  Object {
                    "color": "#394F63",
                  }
                }
              >
                Postcode
              </label>
              <span
                className="card-item-value"
                id="card-item-value"
              >
                
              </span>
            </div>
          </div>
          <div
            className="CardItem inline"
          >
            <div
              className="card-item-inner"
            >
              <label
                className="card-item-label"
                htmlFor="card-item-value"
                style={
                  Object {
                    "color": "#394F63",
                  }
                }
              >
                Country
              </label>
              <span
                className="card-item-value"
                id="card-item-value"
              >
                
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
